### PR TITLE
refactored api end point declarations

### PR DESCRIPTION
**Pull request by vivonk**
## Description

Declared all API endpoints to Server.py, created utils folder for application static functions, constants and more. Created an initializer script which initializes server and socketio at the beginning of socket server starts, removed unnecessary imports from some of the files and changed `e.message` to `str(e)` because `Exception` class does not have any message variable in python 3.x.

## Related Issue

#474 

## How Has This Been Tested?

Run standard tests and checked all the functions from the frontend side, working fine

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by malithsen** [comment]
LGTM

**Comment by vivonk** [comment]
@agentmilindu could you review this so that we can move with webpack things :)

**Comment by vivonk** [comment]
@manushanga using function will also make it repetitive. Yes, errors should be shown in some generic way but on the front end side, we just show `something went wrong` so the user will not distract. Also, if a developer is using Bassa then it would be best that he/she can error in the response itself.

File: components/core/routes/Download.py [hunk 1]
```diff
@@ -1,46 +1,32 @@
...
+from utils.token_utils import token_validator
+from initializer import socketio
 
 
-@server.route('/download/start')
 def start():
 	try:
 		token = request.headers['key']
-		if str(token) != server.config['SECRET_KEY']:
+		if str(token) != server_secret_key:
```
[the remark this answers is not in the document]
Comment (vivonk) [reply components/core/routes/Download.py]: correct

[the remark this answers is not in the document]
Comment (vivonk) [reply components/core/routes/Download.py]: Done @agentmilindu

**Pull request merged**